### PR TITLE
Big refactoring

### PR DESCRIPTION
**Pull request by jsidorenko**

### CODE REVIEW

File: src/components/Pages/Assets/Tokens/LoadTokensData.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  useEffect(() => {
+    getAllTokensWithNativeAndSupply().then(setTokens);
+  }, [getAllTokensWithNativeAndSupply]);
+
+  if (isEmpty(tokens)) {
+    return <>Gathering data... please wait</>;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return <Tokens tokens={tokens!} />;
```
Comment (VadimSaveljev) [inline src/components/Pages/Assets/Tokens/LoadTokensData.tsx]: I thought we came to an agreement on not to use "!", hence the rule even reminds of this.

File: src/contexts/AccountsContext.tsx [hunk 2]
```diff
@@ -49,6 +51,7 @@ export const useAccounts = () => useContext(AccountsContext);
 export const AccountsContextProvider = ({ children }: AccountsContextProviderProps) => {
   const [availableAccounts, setAvailableAccounts] = useState<Account[] | null>(null);
   const [activeAccount, setActiveAccount] = useState<Account | null>(null);
+  const [activeChain, setActiveChain] = useState<Chain | null>(null);
```
Comment (VadimSaveljev) [inline src/contexts/AccountsContext.tsx]: we have storedChain, setStoredChain
why do we need this?

File: src/components/Pages/Assets/Tokens/LoadTokensData.tsx (hunk 1 above)
Comment (jsidorenko) [reply src/components/Pages/Assets/Tokens/LoadTokensData.tsx]: I would say this is an exemption to make our linter happy :)
A couple of lines above, I'm checking the tokens array is not empty `if (isEmpty(tokens)) {` so, it's not necessary to use `!` here, but somehow linter doesn't understand that

Comment (jsidorenko) [reply src/components/Pages/Assets/Tokens/LoadTokensData.tsx]: ok, I found another way of shutting him up

File: src/contexts/AccountsContext.tsx (hunk 2 above)
Comment (jsidorenko) [reply src/contexts/AccountsContext.tsx]: removed

**Review by VadimSaveljev — APPROVED** [review]

**Pull request merged**